src/Public/Async.ps1

```diff
@@ -0,0 +1,1556 @@
+
+.OUTPUTS
+    Returns a boolean value:
+    - $true if the asynchronous route operation exists.
+    - $false if the asynchronous route operation does not exist.
+#>
+function Test-PodeAsyncRouteOperation {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Id
+    )
+    return ($PodeContext.AsyncRoutes.Results.ContainsKey($Id ))
+}
+
+
+<#
+.SYNOPSIS
+    Manages the progress of an asynchronous task within Pode routes.
+
+.DESCRIPTION
+    This function updates the progress of an asynchronous task in Pode. It supports different parameter sets:
+    - StartEnd: Defines progress between a start and end value.
+    - Tick: Increments the progress by a predefined tick value.
+    - TimeBased: Updates progress based on a specified duration and interval.
+    - SetValue: Allows setting the progress to a specific value.
+
+.PARAMETER Start
+    The start value for progress calculation (used in StartEnd parameter set).
+
+.PARAMETER End
+    The end value for progress calculation (used in StartEnd parameter set).
+
+.PARAMETER Steps
+    The number of steps between the start and end values (used in StartEnd parameter set).
+
+.PARAMETER MaxProgress
+    The maximum progress value (default is 100).
+
+.PARAMETER Tick
+    A switch to increment the progress by the predefined tick value.
+
+.PARAMETER UseDecimalProgress
+    A switch to use decimal values for progress.
+
+.PARAMETER IntervalSeconds
+    The interval in seconds for time-based progress updates (default is 5 seconds).
+
+.PARAMETER DurationSeconds
+    The total duration in seconds for time-based progress updates.
+
+.PARAMETER Value
+    The value to set the progress to (used in SetValue parameter set).
+
+.EXAMPLE
+    Set-PodeAsyncProgress -Start 0 -End 100 -Steps 10 -MaxProgress 100
+
+.EXAMPLE
+    Set-PodeAsyncProgress -Tick
+
+.EXAMPLE
+    Set-PodeAsyncProgress -IntervalSeconds 5 -DurationSeconds 300 -MaxProgress 100
+
+.EXAMPLE
+    Set-PodeAsyncProgress -Value 50
+
+.NOTES
+    This function can only be used inside an Async Route Scriptblock in Pode.
+#>
+function Set-PodeAsyncProgress {
+    [CmdletBinding(DefaultParameterSetName = 'StartEnd')]
+    param (
+        [Parameter(Mandatory = $true, ParameterSetName = 'StartEnd')]
+        [double] $Start,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'StartEnd')]
+        [double] $End,
+
+        [Parameter(ParameterSetName = 'StartEnd')]
+        [double] $Steps = 1,
+
+        [Parameter(ParameterSetName = 'TimeBased')]
+        [Parameter(ParameterSetName = 'StartEnd')]
+        [ValidateRange(1, 100)]
+        [double] $MaxProgress = 100,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'Tick')]
+        [switch] $Tick,
+
+        [Parameter(ParameterSetName = 'TimeBased')]
+        [Parameter(ParameterSetName = 'StartEnd')]
+        [Parameter(ParameterSetName = 'SetValue')]
+        [switch] $UseDecimalProgress,
```

Comment:
the idea is that you get 100% only whern the operation is completed. 
The timer method can hit the MaxProgress before the operation completes because the time gave to the operation was not enough.
You don't want to see 100% because the timer was set wrong or the operation took too long.
For me has more sense to see something below 100% like 99% for integer or 99.99% for double.

Regarding Int vs Double I can remove the logic